src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -134,8 +139,8 @@ def _migrate_views(self, acl_strategy, all_grants_to_migrate, all_migrated_group
                     )
                 )
             Threads.strict("migrate views", tasks)
-            self._migration_status_refresher.reset()
             all_tasks.extend(tasks)
+        self._migration_status_refresher.reset()
```

Comment:
@FastLee : You added the migration status refresher. Do you remember why you reset the migration status in-between batches?

Before this change, the migration status has an incosistent outcome depending if there are views to be migrated or not:
- For both, the migration status is reset (line 124) and then indexed (line 126)
- If views to migrate, then the migration status is reset after each batch (also the last)
- If no views, nothing happens  

So the migration status is reset when there views to migrate; it is not when there are no views to migrate

Reply:
> Do you remember why you reset the migration status in-between batches?

It is reset because `self._view_can_be_migrated` uses the migration status to check if dependencies are migrated to decide if a view can be migrated or not. The reset introduces a new `crawl` (not a `fetch` from the inventory database) when the migration status is requested through the `index`.

@FastLee and @nfx : As code improvement // tech debt, I suggest to **always** crawl the migration status and update the migration status in the inventory database. (Not in this PR, but as separate work to be planned accordingly.)

When requesting the migration status in the code (only used in the `table_migrate`), we always want the up-to-date status. Now, we get the up-to-date version by resetting the migration status sometimes, creating implicit dependencies between where the migration status is `reset` and retrieved through `index` - spanning across methods.

It took me quite some time to figure out how/why the migration status was not refreshed in certain cases and why it was in others, and why my first changes let other test fail. IMO the code becomes clearer by removing the implicit dependency.